modules/core/src/matrix_wrap.cpp

```diff
@@ -1045,6 +1045,8 @@ size_t _InputArray::offset(int i) const
     if( k == STD_VECTOR_UMAT )
     {
         const std::vector<UMat>& vv = *(const std::vector<UMat>*)obj;
+        if( i < 0 )
+            return 1;
```

Comment:
Why is 1 here for offset? What is logic here?

I know that this is a copy-paste, but we should not copy mistakes.

IMHO we should ban this method call with negative argument value (through `CV_Assert()`).

Reply:
> Why is 1 here for offset? What is logic here?
I tried to find some design documents about `_InputArray`, but without success.
> I know that this is a copy-paste, but we should not copy mistakes.
> 
> IMHO we should ban this method call with negative argument value (through `CV_Assert()`).
Agree. The current behavior is a bit inconsistent (some kind of matrices raise an error and some return 1). I will fix the patch and tests.